src/server/frontend_wayland/desktop_file_manager.cpp

```diff
@@ -31,6 +31,12 @@ const char* DESKTOP_FILE_POSTFIX = ".desktop";
 std::pair<std::string, std::string> const app_id_to_desktop_file_quirks[] = {
     {"gnome-terminal-server", "org.gnome.Terminal.desktop"} // https://gitlab.gnome.org/GNOME/gnome-terminal/-/issues/8033
 };
+
+inline void rtrim(std::string &s) {
+    s.erase(std::find_if(s.rbegin(), s.rend(), [](unsigned char ch) {
+        return !std::isspace(ch);
+    }).base(), s.end());
```

Comment:
I think a little more whitespace would make this code clearer (more things on different lines), but I'm not going to block for that.